### DISCUSSION TITLE
AboutTheGarden

### DISCUSSION THREAD

**Discussion by giscus** [oldwinter/knowledge-garden#11] [主发布站的评论系统]
# AboutTheGarden

Host your second brain and digital garden for free. Quartz features extremely fast full-text search, Wikilink support, backlinks, local graph, tags, and link previews.

https://garden.oldwinter.top/

**Comment by oldwinter** [comment]
📢 请从其他网站跳转而来的，想看对应文章或笔记的朋友们，在界面右上角搜索文章关键词，一定能找到的。如下图所示，输入『常青笔记』。
原网站使用的是obsidian收费发布服务，现已到期下线，但文章和笔记全部都保留的，无需担心。
<img width="850" alt="image" src="https://user-images.githubusercontent.com/8112159/188563145-2bc9d1d3-d57f-49ac-a9e5-558af5b8954d.png">

**Comment by sakylian** [comment]
good idea! Thank you for your share.

**Comment by shanyang121** [comment]
非常棒的数字花园，受益良多，感谢作者的无偿分享

**Comment by ouyangzhCN** [comment]
hi，感谢发布教程！可以请问一下有什么办法可以支持obsidian的双等号==高亮语法吗？比如添加插件之类的？再次感谢！

**Reply by ouyangzhCN** [reply]
发现原版demo: https://maximevaillancourt.com/blog/setting-up-your-own-digital-garden-with-jekyll
是支持双等号高亮的，请问这边是修改了是哪些地方吗？请教如何改回，谢谢！

**Reply by ouyangzhCN** [reply]
通过比对发现oldwinter修改了 `_plugins/markdown-highlighter.rb` 的第17行
原代码 `doc.content.gsub!(/==+([^ ](.*?)?[^ .=]?)==+/, "<mark>\\1</mark>")`
改为 `doc.content.gsub!(/==+(\w(.*?)?[^ .=]?)==+/, "<mark>\\1</mark>")`
通过修改回去决绝了这个问题

**Reply by oldwinter** [reply]
呃，我倒是没改过这段代码。可能是我fork的是半年前的版本了，期间他们也修了很多bug，估计是已经修复了这个bug。我找个时间把他们的新版本再merge进来~~

**Comment by PinkR1ver** [comment]
👍👍👍👍👍👍👍

请问我在make serve的preview中有interactive graph，但是publish消失了，这是因为什么呢

**Comment by PetrusZ** [comment]
问一下有rss地址吗？

**Reply by oldwinter** [reply]
没有哦，因为是笔记，而且是卡片笔记，不是文章，所以有时候一天几十条笔记，rss意义不是太大的。

**Comment by drinktoomuchsax** [comment]
阿巴阿巴

**Comment by AugustHuangMX** [comment]
作者你好！我是一个小白，我想知道该如何实现本地-网页端的自动更新呀，还是说我每一次都要去终端手动更新一次？

**Comment by oldwinter** [comment]
用电脑端的某些自动化工具，实现git commit，git push操作，即可实现自动更新网页，push后的流程全部是自动完成的。自动化工具比如macos端使用keyboard mastro。<br/><br/><br/><html>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
</head>
<body>
<div id="imail_signature" class="ntes-signature">        <a href="https://dashi.163.com/projects/signature-manager/detail/index.html?ftlId=1&name=cdd2zju%40gmail.com&uid=cdd2zju%40gmail.com&iconUrl=https%3A%2F%2Fmail-online.nosdn.127.net%2Fqiyelogo%2FdefaultAvatar.png&email=%E9%82%AE%E7%AE%B1%EF%BC%9Acdd2zju%40gmail.com&items=%5B%22%E9%82%AE%E7%AE%B1%EF%BC%9Acdd2zju%40gmail.com%22%5D" style="display:block;background:#fff; max-width: 400px; _width: 400px;padding:15px 0 10px 0;text-decoration: none; outline:none;-webkit-tap-highlight-color:transparent;-webkit-text-size-adjust:none !important;text-size-adjust:none !important;">
            <table cellpadding="0" style='width: 100%; max-width: 100%; table-layout: fixed; border-collapse: collapse;color: #9b9ea1;font-size: 14px;line-height:1.3;-webkit-text-size-adjust:none !important;text-size-adjust:none !important;'>
                <tbody style="font-family: 'PingFang SC', 'Hiragino Sans GB','WenQuanYi Micro Hei', 'Microsoft Yahei', '微软雅黑', verdana !important; word-wrap:break-word; word-break:break-all;-webkit-text-size-adjust:none !important;text-size-adjust:none !important;">
                    <tr>
                            <td width="38" style="padding:0; box-sizing: border-box; width: 38px;">
                                <img width="38" height="38" style="vertical-align:middle; width: 38px; height: 38px; border-radius:50%;" src="https://mail-online.nosdn.127.net/qiyelogo/defaultAvatar.png" />
                            </td>
                            <td style='padding: 0 0 0 10px; color: #31353b;'>
                                <div style="font-size: 16px;font-weight:bold; width:100%; white-space: nowrap; overflow:hidden;text-overflow: ***@***.***</div>
                            </td>
                    </tr>
                        <tr width="100%" style="font-size: 14px !important; width: 100%;">
                            <td colspan='2' style="padding:10px 0 0 0; font-size:14px !important; width: 100%;">
                                    <div style="width: 100%;font-size: 14px ***@***.***</div>
                            </td>
                        </tr>
                </tbody>
            </table>
        </a></div><br/><br/><br/><div class="ntes-mailmaster-quote" style="padding-top: 1px; padding-bottom: 1px" >
    <div style=" margin-top: 20px; margin-bottom: 12px; font-size: 14px; line-height: 1.25; color: #89898c; " >---- 回复的原邮件 ----</div>
    <div style=" margin-bottom: 12px; font-size: 13px; line-height: 1.25; color: #2c2c2e; padding: 2px; border-radius: 8px; background-color: #f0f0f0; " >
        <table width="100%" cellpadding="0" cellspacing="10" border="0">
            <tr>
                <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; white-space: nowrap; " >发件人</td>
                <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " ><a class="mail-from" style="color: #1c83eb; text-decoration: none" ***@***.***" >***@***.***&gt;</a></td>
            </tr>
            <tr>
                <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; white-space: nowrap; " >日期</td>
                <td class="mail-date" valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " >2023年04月06日 01:19</td>
            </tr>
            <tr style="">
                <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " >收件人</td>
                <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " ><a class="mail-to" style="color: #1c83eb; text-decoration: none" ***@***.***" >***@***.***&gt;</a></td>
            </tr>
            <tr style="">
                <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " >抄送至</td>
                <td valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " ><a class="mail-cc" style="color: #1c83eb; text-decoration: none" ***@***.***" >DongDong ***@***.***&gt;</a>、<a class="mail-cc" style="color: #1c83eb; text-decoration: none" ***@***.***" >***@***.***&gt;</a></td>
            </tr>
            <tr>
                <td valign="top" style=" width: 4em; font-size: 13px; line-height: 1.25; color: #89898c; " >主题</td>
                <td class="mail-subject" valign="top" style=" font-size: 13px; line-height: 1.25; color: #2c2c2e; word-break: break-all; " >Re: [oldwinter/knowledge-garden] AboutTheGarden (Discussion #11)</td>
            </tr>
        </table>
    </div>
    <div><p></p>
<p dir="auto">作者你好！我是一个小白，我想知道该如何实现本地-网页端的自动更新呀，还是说我每一次都要去终端手动更新一次？</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />Reply to this email directly, <a href="https://github.com/oldwinter/knowledge-garden/discussions/11#discussioncomment-5535111">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/AB54QH3DEO2FBN3MBSDQQZ3W7WSSJANCNFSM57CNRSEA">unsubscribe</a>.<br />You are receiving this because you commented.<img src="https://github.com/notifications/beacon/AB54QH25WYBQ4ASY7ICYXFLW7WSSJA5CNFSM57CNRSEKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOABKHLBY.gif" height="1" width="1" alt="" /><span style="color: transparent; font-size: 0; display: none; visibility: hidden; overflow: hidden; opacity: 0; width: 0; height: 0; max-width: 0; max-height: 0; mso-hide: all">Message ID: <span>&lt;oldwinter/knowledge-garden/repo-discussions/11/comments/5535111</span><span>@</span><span>github</span><span>.</span><span>com&gt;</span></span></p>
<script type="application/ld+json">[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/oldwinter/knowledge-garden/discussions/11#discussioncomment-5535111",
"url": "https://github.com/oldwinter/knowledge-garden/discussions/11#discussioncomment-5535111",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]</script></div>
</div>
</body>
</html>

**Comment by AugustHuangMX** [comment]
了解了！还想知道如何避免每一次手动输入
```
--- 
title: xxx
tags: xxx
---
```
感觉好麻烦，但是如果不这么做，每一页不会显示我在 obsidian 的标题

**Comment by oldwinter** [comment]
我是用linter插件，自动添加这2个metadata字段的。

在 2023年4月6日星期四，AugustHuangMX ***@***.***> 写道：

**Comment by wanghusw** [comment]
你好，请问用 quartz 方案，是按照作者的步骤来吗？我按照他的文档进展到 https://quartz.jzhao.xyz/notes/preview-changes/ 这里的 # Installing Hugo  就不知道怎么进行下去了，`make serve` 命令并没有起作用，可以帮忙吗？

**Reply by wanghusw** [reply]
已经自行解决，用了 Jekyll 方案，您的网站很棒。

**Reply by wanghusw** [reply]
quartz 方案也已经部署成功，再次感谢

**Comment by eonun** [comment]
最近笔记是怎么实现的

**Reply by wanghusw** [reply]
我也想问这个问题。
